### ISSUE TITLE
ATS 8.0.x/8.1.x leaks HTTP/2 connections

### ISSUE THREAD

**Issue by jvgutierrez** [apache/trafficserver#6671]
we've been seeing this consistently across every ATS instance that handles HTTP/2 traffic:
![image](https://user-images.githubusercontent.com/4091505/79537442-c6038280-8071-11ea-938d-468c0fc35bb8.png)
```
vgutierrez@cp1085:~$ sudo -i traffic_ctl --run-root=/srv/trafficserver/tls metric match current_active_client_connections
proxy.process.http.current_active_client_connections 39
proxy.process.http.websocket.current_active_client_connections 0
proxy.process.http2.current_active_client_connections 1597
vgutierrez@cp1085:~$ sudo -i traffic_ctl --run-root=/srv/trafficserver/tls metric match transactions |egrep "current_(client|server)_transactions"
proxy.process.http.current_client_transactions 54
proxy.process.http.current_server_transactions 50
```

`proxy.process.http2.current_active_client_connections` only goes back to 0 if TS is restarted, and we don't believe that's only a metrics issue. Even on depooled servers we had a huge number of ESTABLISHED connections until we lowered `proxy.config.net.default_inactivity_timeout` from the default value of 86400 seconds to 300 seconds

**Comment by zwoop** [comment]
is this still an issue with 8.1.x or 9.0.x ?

**Comment by jvgutierrez** [comment]
yes, I've tested ATS 8.1.x (commit bfd09aa) and this is still happening:
![image](https://user-images.githubusercontent.com/4091505/80449112-c9b6c500-890d-11ea-86c2-2f2315e0cff7.png)

**Comment by maskit** [comment]
I found a case that `proxy.process.http2.current_active_client_connections` is not decremented. However, it seems like the connection is closed on the case.

It happens on master when ATS gracefully closes an H2 connection. There are two places that call `clear_session_active`, which decrements the metric, but the one in `release_stream` is skipped because of graceful shutdown, and the other in `Http2ClientSession::do_io_close` is also not called because `do_io_close` itself is not called.

Like I said, the case I found is only a metric issue, but something is obviously broken. I'll let you know if I find any.

**Comment by jvgutierrez** [comment]
we've tested #6729 and it doesn't fix the issue, stats look basically the same.

**Comment by maskit** [comment]
@jvgutierrez Do you still see the increase? This issue was automatically closed by GitHub when #6876 was merged, but I don't see any comments from you. Please reopen if this still happens.

**Comment by jvgutierrez** [comment]
we backported #6876 to our 8.0.8-1wm2 and it solved the issue for us:
![image](https://user-images.githubusercontent.com/4091505/91146293-38324680-e6a6-11ea-82af-28d100d02d5f.png)
